web/src/pages/Home/useMenuDeleteCallback.tsx

```diff
@@ -1,16 +1,19 @@
 import {useCallback} from 'react';
-import {TTest} from '../../types/Test.types';
-import {useDeleteTestByIdMutation} from '../../redux/apis/TraceTest.api';
-import TestAnalyticsService from '../../services/Analytics/TestAnalytics.service';
+import {useNavigate} from 'react-router-dom';
+import {TTest} from 'types/Test.types';
+import {useDeleteTestByIdMutation} from 'redux/apis/TraceTest.api';
+import TestAnalyticsService from 'services/Analytics/TestAnalytics.service';
 
 export function useMenuDeleteCallback(): (test: TTest) => void {
   const [deleteTestMutation] = useDeleteTestByIdMutation();
+  const navigate = useNavigate();
 
   return useCallback(
     ({id: testId}: TTest) => {
       TestAnalyticsService.onDeleteTest();
       deleteTestMutation({testId});
+      navigate('/');
```

Comment:
Do we need the navigate here?

Reply:
Yeah, if the delete is called from the test details we have to move the user to the home page and if its from the home page then it wont affect the navigation.